app/src/main/java/com/example/concordia_campus_guide/LocationFragment/LocationFragmentViewModel.java

```diff
@@ -1,31 +1,181 @@
+     * @param map is the map used in our application.
+     * @param applicationContext is the Context of the LocationFragmentView page
+     * @return the initiated layer or it will throw an exception if it didn't find the
+     *  GeoJson File
+     */
+    private GeoJsonLayer initLayer(GoogleMap map, Context applicationContext){
+        GeoJsonLayer layer = null;
+        try {
+            layer = new GeoJsonLayer(map, R.raw.buildingcoordinates, applicationContext);
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+        return layer;
     }
 
 
+    /**
+     * @param layer the GeoJson layer containing features to style.
+     * @param map the google map where layer will be displayed and markers will be added.
+     */
+    private void setPolygonStyle(GeoJsonLayer layer, GoogleMap map){
+        for (GeoJsonFeature feature : layer.getFeatures()){
+            feature.setPolygonStyle(getPolygonStyle());
+            String[] coordinates = feature.getProperty("center").split(", ");
+            LatLng centerPos = new LatLng(parseDouble(coordinates[1]), parseDouble(coordinates[0]));
+            addMarker(map, centerPos, Enum.valueOf(BuildingCode.class, feature.getProperty("code")));
+        }
+    }
+
+    /**
+     * The purpose of this method is to add a marker on the specified building.
+     * @param map is the map used in our application.
+     * @param centerPos is the latitude and longitude of the building's center
+     * @param buildingCode is the Building on which the method will add a marker
+     */
+    private void addMarker(GoogleMap map, LatLng centerPos, BuildingCode buildingCode) {
+        Marker marker = map.addMarker(
+                new MarkerOptions()
+                        .position(centerPos)
+                        .icon(BitmapDescriptorFactory.fromResource(getIcon(buildingCode)))
+                        .flat(true)
+                        .anchor(0.5f,0.5f)
+                        .alpha(0.90f)
+                        .title(buildingCode.toString())
+        );
+        marker.setTag(buildingCode);
+    }
+
+    /**
+     * The purpose of this method is the polygons style after setting their
+     * FillColor, StrokeColor and StrokeWidth
+     * @return it returns the polygon style.
+     */
+    public GeoJsonPolygonStyle getPolygonStyle() {
+        GeoJsonPolygonStyle geoJsonPolygonStyle = new GeoJsonPolygonStyle();
+        geoJsonPolygonStyle.setFillColor(Color.argb(51, 18, 125, 159));
+        geoJsonPolygonStyle.setStrokeColor(Color.argb(255, 18, 125, 159));
+        geoJsonPolygonStyle.setStrokeWidth(6);
+        return geoJsonPolygonStyle;
+    }
+
+
+    /**
+     * @param buildingCode it represents which building we will be covering
+     * @return Int of drawable resource's bitmap representation
+     */
+    public int getIcon(BuildingCode buildingCode){
+        switch (buildingCode){
```

Comment:
Is there a way to programmatically get the resource, rather than having this huge switch statement? For example, I stored my images in assets, and I can access them this way: https://github.com/Concordia-Campus-Guide/Concordia-Campus-Guide/blob/5bfafbb19bdea5f2585bbc172a69bf097d812dc4/app/src/main/java/com/example/concordia_campus_guide/InfoCardFragment/InfoCardFragment.java#L70

Reply:
Why did you create an assets folder ? It is providing the same services as the res folder.

Reply:
because res is not meant to be accessed programmatically, and I don't think it is good to create a very long switch statement to return the correct res

Reply:
```So why have an assets folder at all? If you want to compute the asset you want to use at run time, it's pretty easy. With resources, you would have to declare a list of all the resource IDs that might be used and compute an index into the the list. (This is kind of awkward and introduces opportunities for error if the set of resources changes in the development cycle.) (EDIT: you can retrieve a resource ID by name using getIdentifier, but this loses the benefits of compile-time checking.) Assets can also be organized into a folder hierarchy, which is not supported by resources. It's a different way of managing data. Although resources cover most of the cases, assets have their occasional use.```

https://stackoverflow.com/questions/5583608/difference-between-res-and-assets-directories

Reply:
I also think it's useful to have subfolders, because too many files will easily get lost

Reply:
coming back to my original question (we can continue the assets vs res discussion elsewhere), do you think it is good practice to create a very long switch in order to use the res?

Reply:
Ok. that's a reasonable point. I am just curious. For your part, did you create a sub-folder in the assets ?

Reply:
I agree with you. It is not. @tarekait1996  had a similar comment about that.

Reply:
yes, for my part, I have a sub-folder in the assets `buildings_images`

Reply:
Hi, the reason I had put the images in the res folder is that the map markers can only use bitmap images as icons, and that was the only option I could find to convert png images to bitmap easily. From what I looked up every res in the drawable folder will have a bitmap representation of it. Furthermore, that was also the way that was shown in the google map docs for adding custom icons.